tests/rados/test_norecovery_flag_functionality.py

```diff
@@ -38,195 +38,288 @@ def run(ceph_cluster, **kw):
         pool_obj.rm_bulk_flag(pool_name=pool_name)
         rados_obj.log_cluster_health()
         rados_obj.bench_write(
             pool_name=pool_name, byte_size=1024, rados_write_duration=10
         )
-        rados_obj.log_cluster_health()
-        status = check_status(pool_obj, old_pg_num, pool_name)
-        pg_num = pool_obj.get_pg_autoscaler_value(pool_name, "pg_num_target")
-        if not status:
+
+        if not check_status(pool_obj, old_pg_num, pool_name):
             log.error(
-                f"Scenario2:The pg number is decreased to after setting the norecover flag "
-                f"-{pg_num}"
+                "Scenario2:The pg number is decreased to after "
+                "setting the norecover flag on the pool with bulk flag removed"
             )
             return 1
         log.info(
-            f"Scenario2:The pg number is not decreased after setting the norecover flag.The pg_num is-{pg_num}"
+            "Scenario2:The pg number is not decreased after setting the norecover flag."
+            "Removing the bulk flag on the pool to check if the PG count would increase now"
         )
+
         utils.configure_osd_flag(ceph_cluster, "unset", "norecover")
         log.info("The norecover is unset on the cluster")
-        status = check_status(pool_obj, pg_num, pool_name)
-        pg_num = pool_obj.get_pg_autoscaler_value(pool_name, "pg_num_target")
-        if status:
-            expected_pg_num = pool_obj.get_pg_autoscaler_value(
-                pool_name, "pg_num_final"
+
+        if check_status(pool_obj, old_pg_num, pool_name):
+            log.error(
+                "Scenario2:The pg number is not decreased to after unsetting the norecover flag."
+                "on the pool where bulk flag was removed"
             )
-            log.info(f"The expected pg number is -{expected_pg_num}")
-            if pg_num != expected_pg_num:
-                log.error(
-                    f"Scenario2:The pg number is not decreased to after unsetting the norecover flag."
-                    f"The pg number is -{pg_num}"
-                )
-                return 1
+            return 1
+
+        if not check_pg_increase(pool_name=pool_name):
+            log.error(
+                "PG count not increased on the pool upon creation of pool with bulk flag"
+                "Test Failed"
+            )
+            return 1
         log.info(
-            f"Scenario2:The pg number is decreased after unsetting the norecover flag."
-            f"The pg number is -{pg_num}"
+            "Scenario2:The pg number is decreased after unsetting the norecover"
+            " flag upon bulk flag addition/ removal complete"
         )
         delete_and_healthCheck(rados_obj, pool_name)
         log.info("=========Scenario2: Test completed===========")
+        time.sleep(20)
+
         log.info(
             "Scenario3:Modify the pg number manually and verify that the pg_num should not increase "
```

Comment:
Following are the manual steps of scenario-3 and 4 which are automated -
ceph osd set norecover
ceph osd pool set  test pg_num 8
Check the “ceph osd pool autoscale-status” .The pg_num should not be increased
ceph osd unset norecover
**Verification step**- The pg_num should be increased. 
ceph osd pool set  test pg_num 512
ceph osd set norecover
rados bench -p test 5 write -b 1024 --no-cleanup
**Verification Step** -Check the “ceph osd pool autoscale-status” .The pg_num should not be decreased
ceph osd unset norecover

Modified Code is(From bugzilla) -
1. Set norecovery flag on the cluster
 ceph osd set norecover
2. Create pool with 8 PGs
   ceph osd pool create test 8
3. As expected, the autoscaling of the PGs is not happening.
4. Change the PG count on the pool. See that PGs start scaling. 
cmd : ceph osd pool set <pool> pg_num 256

We can add this as a new scenario.

Reply:
> Following are the manual steps of scenario-3 and 4 which are automated - ceph osd set norecover ceph osd pool set test pg_num 8 Check the “ceph osd pool autoscale-status” .The pg_num should not be increased ceph osd unset norecover 
-> this is already covered as part of scenario 1.

**Verification step**- The pg_num should be increased. ceph osd pool set test pg_num 512 ceph osd set norecover rados bench -p test 5 write -b 1024 --no-cleanup 
**Verification Step** -Check the “ceph osd pool autoscale-status” .The pg_num should not be decreased ceph osd unset norecover
-> this is already covered as part of scenario 3, Which is asked about in the bugzilla. 

> 
> Modified Code is(From bugzilla) -
> 
> 1. Set norecovery flag on the cluster
>    ceph osd set norecover
> 2. Create pool with 8 PGs
>    ceph osd pool create test 8
> 3. As expected, the autoscaling of the PGs is not happening.
> 4. Change the PG count on the pool. See that PGs start scaling.
>    cmd : ceph osd pool set  pg_num 256
> 
> We can add this as a new scenario.